src/app/clusters/power-topology-server/PowerTopologyCluster.cpp

```diff
@@ -39,91 +39,87 @@ namespace app {
 namespace Clusters {
 namespace PowerTopology {
 
-CHIP_ERROR Instance::Init()
-{
-    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INCORRECT_STATE);
-    return CHIP_NO_ERROR;
-}
-
-void Instance::Shutdown()
+CHIP_ERROR PowerTopologyCluster::GetAvailableEndpoints(AttributeValueEncoder & aEncoder) const
 {
-    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
-}
-
-bool Instance::HasFeature(Feature aFeature) const
-{
-    return mFeature.Has(aFeature);
-}
-
-bool Instance::SupportsOptAttr(OptionalAttributes aOptionalAttrs) const
-{
-    return mOptionalAttrs.Has(aOptionalAttrs);
-}
-
-// AttributeAccessInterface
-CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
-{
-    switch (aPath.mAttributeId)
-    {
-    case FeatureMap::Id:
-        ReturnErrorOnFailure(aEncoder.Encode(mFeature));
-        break;
-    case AvailableEndpoints::Id:
-        return ReadAvailableEndpoints(aEncoder);
-    case ActiveEndpoints::Id:
-        return ReadActiveEndpoints(aEncoder);
-    }
-    return CHIP_NO_ERROR;
-}
-
-CHIP_ERROR Instance::ReadAvailableEndpoints(AttributeValueEncoder & aEncoder)
-{
-    if (!SupportsOptAttr(OptionalAttributes::kOptionalAttributeAvailableEndpoints))
-    {
-        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
-    }
-    VerifyOrReturnError(HasFeature(Feature::kSetTopology), CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE,
+    VerifyOrReturnError(mFeatureFlags.Has(Feature::kSetTopology), CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE,
                         ChipLogError(Zcl, "Power Topology: can not get AvailableEndpoints, feature is not supported"));
-
     return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
         for (uint8_t i = 0; true; i++)
         {
             EndpointId endpointId;
             auto err = mDelegate.GetAvailableEndpointAtIndex(i, endpointId);
-            if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
-            {
-                return CHIP_NO_ERROR;
-            }
+
+            VerifyOrReturnError(!(err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED), CHIP_NO_ERROR); // End of list, safe to exit normally
```

Comment:
```suggestion
            VerifyOrReturnError(err != CHIP_ERROR_PROVIDER_LIST_EXHAUSTED, CHIP_NO_ERROR); // End of list, safe to exit normally
```

because `!(a == b)` is strange...